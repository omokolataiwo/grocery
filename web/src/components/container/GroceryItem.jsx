import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Grocery = ({
  grocery, deleteItem, startEditing, toggleBuy
}) => (
  <div className="grocery-item col-md-4">
    <h2 className="title">{grocery.name}</h2>
    <p className="amount">
      &#8358;
      {grocery.amount}
    </p>

    <div className="control">
      <FontAwesomeIcon
        onClick={() => deleteItem(grocery._id)}
        onKeyUp={() => deleteItem(grocery._id)}
        role="button"
        tabIndex="-99999"
        icon="trash"
        id={`delete-${grocery._id}`}
        style={{ color: '#ff8f9a' }}
      />

      <FontAwesomeIcon
        onClick={() => startEditing(grocery)}
        onKeyUp={() => startEditing(grocery)}
        tabIndex="-199999"
        role="button"
        icon="edit"
        style={{ color: 'rgb(119, 172, 241)' }}
        id={`edit-${grocery._id}`}
      />

      <FontAwesomeIcon
        onClick={() => toggleBuy(grocery._id)}
        onKeyUp={() => toggleBuy(grocery._id)}
        tabIndex="-199999"
        role="button"
        id={`buy-${grocery._id}`}
        style={{ color: '#28a745' }}
        icon={grocery.purchased ? 'cart-plus' : 'shopping-cart'}
      />
    </div>
  </div>
);

Grocery.propTypes = {
  grocery: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    amount: PropTypes.string.isRequired
  }).isRequired,
  deleteItem: PropTypes.func.isRequired,
  toggleBuy: PropTypes.func.isRequired,
  startEditing: PropTypes.func.isRequired
};

export default Grocery;
