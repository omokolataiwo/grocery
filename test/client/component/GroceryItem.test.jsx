import React from 'react';
import { shallow } from 'enzyme';
import GroceryItem from '../../../web/src/components/container/GroceryItem';

const props = {
  grocery: {
    _id: 'msken',
    name: 'Bread',
    amount: '',
    purchased: false
  },
  errors: {},
  deleteItem: jest.fn(() => {}),
  startEditing: jest.fn(() => {}),
  toggleBuy: jest.fn(() => {})
};
const wrapper = shallow(<GroceryItem {...props} />);

describe('AddGroceryItem Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should buy item', () => {
    const btnBuy = wrapper.find('#buy-msken');
    btnBuy.simulate('click');
    btnBuy.simulate('keyup');
    expect(props.toggleBuy).toHaveBeenCalled();
  });

  it('should buy item', () => {
    const btnEdit = wrapper.find('#edit-msken');
    btnEdit.simulate('click');
    btnEdit.simulate('keyup');
    expect(props.startEditing).toHaveBeenCalled();
  });

  it('should delete item', () => {
    const btnDelete = wrapper.find('#delete-msken');
    btnDelete.simulate('click');
    btnDelete.simulate('keyup');
    expect(props.deleteItem).toHaveBeenCalled();
  });
});
