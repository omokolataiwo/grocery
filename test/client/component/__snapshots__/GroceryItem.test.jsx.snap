// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddGroceryItem Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grocery
    deleteItem={[MockFunction]}
    errors={Object {}}
    grocery={
      Object {
        "_id": "msken",
        "amount": "",
        "name": "Bread",
        "purchased": false,
      }
    }
    startEditing={[MockFunction]}
    toggleBuy={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
        >
          Bread
        </h2>,
        <p
          className="amount"
        >
          ₦
          
        </p>,
        <div
          className="control"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="trash"
            id="delete-msken"
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            onKeyUp={[Function]}
            pull={null}
            pulse={false}
            role="button"
            rotation={null}
            size={null}
            spin={false}
            style={
              Object {
                "color": "#ff8f9a",
              }
            }
            symbol={false}
            tabIndex="-99999"
            title=""
            transform={null}
          />
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="edit"
            id="edit-msken"
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            onKeyUp={[Function]}
            pull={null}
            pulse={false}
            role="button"
            rotation={null}
            size={null}
            spin={false}
            style={
              Object {
                "color": "rgb(119, 172, 241)",
              }
            }
            symbol={false}
            tabIndex="-199999"
            title=""
            transform={null}
          />
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="shopping-cart"
            id="buy-msken"
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            onKeyUp={[Function]}
            pull={null}
            pulse={false}
            role="button"
            rotation={null}
            size={null}
            spin={false}
            style={
              Object {
                "color": "#28a745",
              }
            }
            symbol={false}
            tabIndex="-199999"
            title=""
            transform={null}
          />
        </div>,
      ],
      "className": "grocery-item col-md-4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bread",
          "className": "title",
        },
        "ref": null,
        "rendered": "Bread",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "₦",
            "",
          ],
          "className": "amount",
        },
        "ref": null,
        "rendered": Array [
          "₦",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="trash"
              id="delete-msken"
              inverse={false}
              listItem={false}
              mask={null}
              onClick={[Function]}
              onKeyUp={[Function]}
              pull={null}
              pulse={false}
              role="button"
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "#ff8f9a",
                }
              }
              symbol={false}
              tabIndex="-99999"
              title=""
              transform={null}
            />,
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="edit"
              id="edit-msken"
              inverse={false}
              listItem={false}
              mask={null}
              onClick={[Function]}
              onKeyUp={[Function]}
              pull={null}
              pulse={false}
              role="button"
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "rgb(119, 172, 241)",
                }
              }
              symbol={false}
              tabIndex="-199999"
              title=""
              transform={null}
            />,
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="shopping-cart"
              id="buy-msken"
              inverse={false}
              listItem={false}
              mask={null}
              onClick={[Function]}
              onKeyUp={[Function]}
              pull={null}
              pulse={false}
              role="button"
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "#28a745",
                }
              }
              symbol={false}
              tabIndex="-199999"
              title=""
              transform={null}
            />,
          ],
          "className": "control",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "fixedWidth": false,
              "flip": null,
              "icon": "trash",
              "id": "delete-msken",
              "inverse": false,
              "listItem": false,
              "mask": null,
              "onClick": [Function],
              "onKeyUp": [Function],
              "pull": null,
              "pulse": false,
              "role": "button",
              "rotation": null,
              "size": null,
              "spin": false,
              "style": Object {
                "color": "#ff8f9a",
              },
              "symbol": false,
              "tabIndex": "-99999",
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "fixedWidth": false,
              "flip": null,
              "icon": "edit",
              "id": "edit-msken",
              "inverse": false,
              "listItem": false,
              "mask": null,
              "onClick": [Function],
              "onKeyUp": [Function],
              "pull": null,
              "pulse": false,
              "role": "button",
              "rotation": null,
              "size": null,
              "spin": false,
              "style": Object {
                "color": "rgb(119, 172, 241)",
              },
              "symbol": false,
              "tabIndex": "-199999",
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "fixedWidth": false,
              "flip": null,
              "icon": "shopping-cart",
              "id": "buy-msken",
              "inverse": false,
              "listItem": false,
              "mask": null,
              "onClick": [Function],
              "onKeyUp": [Function],
              "pull": null,
              "pulse": false,
              "role": "button",
              "rotation": null,
              "size": null,
              "spin": false,
              "style": Object {
                "color": "#28a745",
              },
              "symbol": false,
              "tabIndex": "-199999",
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
          >
            Bread
          </h2>,
          <p
            className="amount"
          >
            ₦
            
          </p>,
          <div
            className="control"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="trash"
              id="delete-msken"
              inverse={false}
              listItem={false}
              mask={null}
              onClick={[Function]}
              onKeyUp={[Function]}
              pull={null}
              pulse={false}
              role="button"
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "#ff8f9a",
                }
              }
              symbol={false}
              tabIndex="-99999"
              title=""
              transform={null}
            />
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="edit"
              id="edit-msken"
              inverse={false}
              listItem={false}
              mask={null}
              onClick={[Function]}
              onKeyUp={[Function]}
              pull={null}
              pulse={false}
              role="button"
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "rgb(119, 172, 241)",
                }
              }
              symbol={false}
              tabIndex="-199999"
              title=""
              transform={null}
            />
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="shopping-cart"
              id="buy-msken"
              inverse={false}
              listItem={false}
              mask={null}
              onClick={[Function]}
              onKeyUp={[Function]}
              pull={null}
              pulse={false}
              role="button"
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "#28a745",
                }
              }
              symbol={false}
              tabIndex="-199999"
              title=""
              transform={null}
            />
          </div>,
        ],
        "className": "grocery-item col-md-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bread",
            "className": "title",
          },
          "ref": null,
          "rendered": "Bread",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "₦",
              "",
            ],
            "className": "amount",
          },
          "ref": null,
          "rendered": Array [
            "₦",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="trash"
                id="delete-msken"
                inverse={false}
                listItem={false}
                mask={null}
                onClick={[Function]}
                onKeyUp={[Function]}
                pull={null}
                pulse={false}
                role="button"
                rotation={null}
                size={null}
                spin={false}
                style={
                  Object {
                    "color": "#ff8f9a",
                  }
                }
                symbol={false}
                tabIndex="-99999"
                title=""
                transform={null}
              />,
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="edit"
                id="edit-msken"
                inverse={false}
                listItem={false}
                mask={null}
                onClick={[Function]}
                onKeyUp={[Function]}
                pull={null}
                pulse={false}
                role="button"
                rotation={null}
                size={null}
                spin={false}
                style={
                  Object {
                    "color": "rgb(119, 172, 241)",
                  }
                }
                symbol={false}
                tabIndex="-199999"
                title=""
                transform={null}
              />,
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="shopping-cart"
                id="buy-msken"
                inverse={false}
                listItem={false}
                mask={null}
                onClick={[Function]}
                onKeyUp={[Function]}
                pull={null}
                pulse={false}
                role="button"
                rotation={null}
                size={null}
                spin={false}
                style={
                  Object {
                    "color": "#28a745",
                  }
                }
                symbol={false}
                tabIndex="-199999"
                title=""
                transform={null}
              />,
            ],
            "className": "control",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": "trash",
                "id": "delete-msken",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "onClick": [Function],
                "onKeyUp": [Function],
                "pull": null,
                "pulse": false,
                "role": "button",
                "rotation": null,
                "size": null,
                "spin": false,
                "style": Object {
                  "color": "#ff8f9a",
                },
                "symbol": false,
                "tabIndex": "-99999",
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": "edit",
                "id": "edit-msken",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "onClick": [Function],
                "onKeyUp": [Function],
                "pull": null,
                "pulse": false,
                "role": "button",
                "rotation": null,
                "size": null,
                "spin": false,
                "style": Object {
                  "color": "rgb(119, 172, 241)",
                },
                "symbol": false,
                "tabIndex": "-199999",
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": "shopping-cart",
                "id": "buy-msken",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "onClick": [Function],
                "onKeyUp": [Function],
                "pull": null,
                "pulse": false,
                "role": "button",
                "rotation": null,
                "size": null,
                "spin": false,
                "style": Object {
                  "color": "#28a745",
                },
                "symbol": false,
                "tabIndex": "-199999",
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
